### PR TITLE
fix(mongo): sorting with UnderscoreNamingStrategy

### LINKED ISSUE #4313: orderBy is not working with UnderscoreNamingStrategy in Mongodb
I use MongoDB with UnderscoreNamingStrategy  
`find` and `findOne` with sort by camel case field do not work, sorting is ignored.  

The problem seems to be exactly with Mongo driver.

**To Reproduce**
```
@Entity()
export class A {
  @PrimaryKey()
  _id = randomUUID();
  
  @Property()
  complexName!: string;
}

orm = await MikroORM.init({
  entities: [A],
  clientUrl: await initMongoReplSet('mikro-orm-test'),
  driver: MongoDriver,
  namingStrategy: UnderscoreNamingStrategy,
});
await orm.schema.clearDatabase()

const a = new A();
a.complexName = 'a';
const b = new A();
b.complexName = 'b';
await orm.em.persistAndFlush(a);
await orm.em.persistAndFlush(b);
orm.em.clear();

const data2 = await orm.em.find(A, {}, {
  orderBy: { complexName: 'desc' },
});
expect(data2[0].complexName).toBe('b');
```
**Expected behavior**
Keys in `orderBy` should be transformed according to naming strategy.

**Versions**

| Dependency | Version |
| - | - |
| node | 18.14 |
| typescript | 4.9.5 |
| mikro-orm | 5 |
| your-driver | MongoDB |

### PR DESCRIPTION
**Pull request by funduck**
Fixes #4313 

I propose to apply `this.renameFields` to `options.orderBy` the same way it is used for `where`

### CODE REVIEW

File: tests/issues/GH4313.test.ts [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+}
+
+describe('GH issue 4313', () => {
+
+  describe('Basic naming strategy with Mongo', () => {
+
+    it('Init orm with basic naming strategy',async () => {
+      orm = await MikroORM.init({
+        entities: [A],
+        clientUrl: await initMongoReplSet('mikro-orm-test'),
```
Comment (B4nan) [inline tests/issues/GH4313.test.ts:85]: no need for replica sets, that memory server implementation is rather unstable and i am using it only in places that are actually testing transactions

File: tests/issues/GH4313.test.ts [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+
+    const data4 = await orm.em.findOne(A, { complexName: { $ne: null } }, {
+      orderBy: [{ complexName: 'desc' }],
+    });
+    expect(data4!.complexName).toBe('b');
+
+  });
+}
+
+describe('GH issue 4313', () => {
```
Comment (B4nan) [inline tests/issues/GH4313.test.ts:78]: no need for nested describe blocks - actually no need for any of them, use just one `test` block on top level

File: tests/issues/GH4313.test.ts [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+      });
+      await orm.schema.clearDatabase();
+    });
+
+    testCase();
+
+    it('Close connection', async () => {
+      await orm.close();
+      await closeReplSets();
+    });
```
Comment (B4nan) [inline tests/issues/GH4313.test.ts:117]: this should be `afterAll` hook, not a separate test case

File: tests/issues/GH4313.test.ts [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+
+    it('Init orm with underscore naming strategy',async () => {
+      orm = await MikroORM.init({
+        entities: [A],
+        clientUrl: await initMongoReplSet('mikro-orm-test'),
+        driver: MongoDriver,
+        namingStrategy: UnderscoreNamingStrategy,
+      });
+      await orm.schema.clearDatabase();
+    });
```
Comment (B4nan) [inline tests/issues/GH4313.test.ts:110]: this should be `beforeAll` hook, not a separate test case

File: tests/issues/GH4313.test.ts [hunk 5]
```diff
@@ -0,0 +1,121 @@
...
+
+  });
+
+  describe('Underscore naming strategy with Mongo', () => {
+
+    it('Init orm with underscore naming strategy',async () => {
+      orm = await MikroORM.init({
+        entities: [A],
+        clientUrl: await initMongoReplSet('mikro-orm-test'),
```
Comment (B4nan) [inline tests/issues/GH4313.test.ts:105]: keep the defaults, just set the database name via `dbName`

File: tests/issues/GH4313.test.ts [hunk 6]
```diff
@@ -0,0 +1,121 @@
...
+function testCase() {
+  test('Insert entities', async () => {
+    const a = new A();
+    a.complexName = 'a';
+    const b = new A();
+    b.complexName = 'b';
+    await orm.em.persistAndFlush(a);
+    await orm.em.persistAndFlush(b);
+    orm.em.clear();
+  });
```
Comment (B4nan) [inline tests/issues/GH4313.test.ts:28]: this should be also part of that (single) `beforeAll` hook

File: tests/issues/GH4313.test.ts [hunk 7]
```diff
@@ -0,0 +1,121 @@
...
+let orm: MikroORM<MongoDriver>;
+
+function testCase() {
+  test('Insert entities', async () => {
+    const a = new A();
+    a.complexName = 'a';
+    const b = new A();
+    b.complexName = 'b';
+    await orm.em.persistAndFlush(a);
+    await orm.em.persistAndFlush(b);
```
Comment (B4nan) [inline tests/issues/GH4313.test.ts:26]: you can pass both in a single call
```suggestion
    await orm.em.persistAndFlush([a, b]);
```

File: tests/issues/GH4313.test.ts [hunk 8]
```diff
@@ -0,0 +1,121 @@
...
+
+    const data3 = await orm.em.findOne(A, { complexName: { $ne: null } }, {
+      orderBy: [{ complexName: 'asc' }],
+    });
+    expect(data3!.complexName).toBe('a');
+
+    const data4 = await orm.em.findOne(A, { complexName: { $ne: null } }, {
+      orderBy: [{ complexName: 'desc' }],
+    });
+    expect(data4!.complexName).toBe('b');
```
Comment (B4nan) [inline tests/issues/GH4313.test.ts:73]: i dont think we need to test `asc` vs `desc`, that's kinda irrelevant to the issue, one direction is enough, the problem is about the left side, not right side

File: tests/issues/GH4313.test.ts [hunk 9]
```diff
@@ -0,0 +1,121 @@
...
+  _id = randomUUID();
+
+  @Property()
+  complexName!: string;
+
+}
+
+let orm: MikroORM<MongoDriver>;
+
+function testCase() {
```
Comment (B4nan) [inline tests/issues/GH4313.test.ts:19]: lets inline this

**Review by B4nan — CHANGES_REQUESTED** [review]
thanks! interesting this wasnt reported before, i guess most of the mongo folks dont really use different strategies/field names

left a few comments about the test case, i'd like to see that greatly simplified, but will be happy to do that myself after merging if you'd struggle somehow with that.

File: tests/issues/GH4313.test.ts (hunk 8 above)
Comment (funduck) [reply tests/issues/GH4313.test.ts:73]: one case may pass just by an accident, but both wont

File: tests/issues/GH4313.test.ts (hunk 2 above)
Comment (funduck) [reply tests/issues/GH4313.test.ts:78]: Ok, I'll make it shorter

File: tests/issues/GH4313.test.ts (hunk 1 above)
Comment (funduck) [reply tests/issues/GH4313.test.ts:85]: Wow, how much faster the test have become without replset! Thanks for hint!

**Comment by funduck** [comment]
Fixed most of the issues you've mentioned

**Review by B4nan — APPROVED** [review]

**Pull request merged**